### DISCUSSION TITLE
Request: support asar archives

### DISCUSSION THREAD

**Discussion by JonnyTech** [doublecmd/doublecmd#1941] [Suggestions]
Please add support for asar archives

https://github.com/electron/asar

~~~
Asar is a simple extensive archive format, it works like tar that concatenates all files together without compression, while having random access support.
Features

    Support random access
    Use JSON to store files' information
    Very easy to write a parser
~~~

**Comment by j2969719** [comment]
```ini
[asar]
Archiver=asar
Description=asar (read-only)
Extension=asar
Format0=?n+
List=%P l %AQ
Extract=%P e %AQ .
Enabled=1
```
tbh for multiarc asar is a pain in the ass because when packing it only accepts a folder and an exclusion pattern, and for a list it only returns file names

**Reply by alexx2000** [reply]
Yes, it is better to write a wcx-plugin. Archive format is not hard.

**Reply by JonnyTech** [reply]
Wow, thanks for the fast reply. I shall study multiarc and find the documentation as that looks extremely flexible. As for wcx-plugin, is there a sdk available?

**Reply by alexx2000** [reply]
https://github.com/doublecmd/doublecmd/wiki/Plugins-development

**Reply by j2969719** [reply]
with #1937 and #1935 the packing and list will work more or less fine (`%P p . %AQ`), but both of them are two big asterisks
imho wcx is overkill for this, the only benefit is that its possible to show filesizes, there doesnt seem to be other anything useful for wcx stored in that json anyway
